hdeps/cli.py

```diff
@@ -169,19 +172,54 @@ def main(
         color=ctx.color,
     )
 
-    for dep in deps:
-        walker.feed(Requirement(dep))
-    for req in requirements_file:
-        walker.feed_file(Path(req))
+    def solve() -> None:
+        for dep in deps:
+            walker.feed(Requirement(dep))
+        for req in requirements_file:
+            walker.feed_file(Path(req))
+        walker.drain()
+
+    solve()
 
-    walker.drain()
     if install_order:
         walker.print_flat()
     else:
         if print_legend:
             walker.print_legend()
         walker.print_tree()
 
+    click.echo("========== Summary ==========")
+    if walker.known_conflicts:
+        resolutions: List[Requirement] = []
+        for project, versions in walker.known_conflicts.copy().items():
+            click.echo(
+                f"Found conflict: {project} {sorted([str(x) for x in versions])}"
+            )
+            for version in versions:
+                LOG.debug(f"Trying to pin {project}=={version}")
+                req = Requirement(f"{project}=={version}")
+                walker.clear()
+                walker.feed(req)
```

Comment:
Might be worth a comment mentioning why it goes first (to see if it can be reused by _all_ of the user requirements' deps)